### PR TITLE
WMS-441 | Add Tabs component

### PR DESCRIPTION
**Pull request by PedroHLC**
#### :thinking: What?
Add a new component: Tabs

#### :man_shrugging: Why?
It's used in WMS Phase3 Group's screen.

#### :pushpin: Jira Issue
[WMS-441](https://paacklogistics.atlassian.net/browse/WMS-441)

#### :no_good: Blocked by
Not blocked, based on master.

#### :clipboard: Pending
- [x] Documentation.
- [x] ARIA labels.

### :fire: Extra
--

### CODE REVIEW

**Comment by PedroHLC** [comment]
Preview

File: src/UI/Tabs.elm [hunk 1]
```diff
@@ -0,0 +1,197 @@
...
+{-| A single element in a tab list.
+-}
+type Item msg
+    = Item ItemConfig (ItemAction msg)
+
+
+type alias ItemConfig =
+    { label : String
+    , isCurrent : Bool
+    }
```
[the remark this answers is not in the document]
Comment (PedroHLC) [reply src/UI/Tabs.elm:71]: About both the redundancy and here, there was a previous discussion about the navigator's sidebar. And adding a shared logic here would only constraint use-cases. For example, you can mix messages and links, have complex tabs that depend on more than one field to query its state.

TL;DR: It's some logic for apps and not the UI.

File: showcase/src/Tabs/Stories.elm [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+artistsTabs : RenderConfig -> Stories.TabsDemo -> Element Msg.Msg
+artistsTabs renderConfig selected =
+    Tabs.fromList
+        [ Tabs.cmd "About"
+            (select Stories.About)
+            (selected == Stories.About)
+        , Tabs.cmd "Corey Taylor"
+            (select Stories.CoreyTaylor)
+            (selected == Stories.CoreyTaylor)
+        , Tabs.cmd "Matt Bellamy"
```
[the remark this answers is not in the document]
Comment (PedroHLC) [reply showcase/src/Tabs/Stories.elm:48]: This was modeled before and discarded in a previous discussion. (See the other comment).
But indeed I should cut redundancy here in this module!

**Comment by PedroHLC** [comment]
> Ideally, the user should be able to change tabs by just focusing once and then using the left/right keys. If that requires too much effort, then just having the ability to focus a tab (and press enter) would solve the issue.

Yeah, MDN recommends using the button tag for this read. But I'm not sure will use this feature in Paack as we have a lot of other elements in the screen and using the arrows may become confusing.

I would say to ping Sufyian as a UX expert and if he agrees then create a card and do this in a later PR.

**Comment by renanpvaz** [comment]
Cool so what do you think about just having the tabs as button tags to allow for basic keyboard navigation and we can revisit this later when the tabs start being used?

**Comment by PedroHLC** [comment]
> Cool so what do you think about just having the tabs as button tags to allow for basic keyboard navigation and we can revisit this later when the tabs start being used?

There is no-way for having custom tags with elm-ui content.
Then I just noticed keyboard navigation wasn't working because I had `tabindex` in the wrong element.

Instead, I ended-up using elm-ui's button and letting it handle `tabindex`. Seems to be working with Tab, Shift+Tab, and Space.

File: showcase/src/Tabs/Stories.elm [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        )
+
+
+artistsTabs : RenderConfig -> Stories.TabsDemo -> Element Msg.Msg
+artistsTabs renderConfig selected =
+    Tabs.fromList
+        [ tab selected "About" Stories.About
+        , tab selected "Corey Taylor" Stories.CoreyTaylor
+        , tab selected "Matt Bellamy" Stories.MattBellamy
+        , tab selected "Dave Ghrowl" Stories.DaveGhrowl
```
Comment (jouderianjr) [inline showcase/src/Tabs/Stories.elm:45]: `Grohl` 💩

**Comment by jouderianjr** [comment]
Also, I think would be nice to add some tests for this component.

**Comment by PedroHLC** [comment]
> @jouderianjr: is there a use case for that?

Not in our current codebase. But for future usage it seems interesting, for example, if we move to use links in tabs, like: `/groups/91823/spaces`, and `/groups/1092849/packages`. (I actually think that we should be doing this right now)
WDYT about it?

**Comment by PedroHLC** [comment]
These tests were fun to do! :baby_chick:  They're really testing something this time :smile:
